src/main/kotlin/com/manimdsl/shapes/Shapes.kt

```diff
@@ -19,44 +20,47 @@ sealed class Shape {
     }
 }
 
-sealed class ShapeWithText : Shape() {
-    // Python shape classes which has shape and text members to manipulate color
+sealed class ShapeWithText : Shape()
 
-    fun getTextMObject(): String {
-        return "$ident.text"
-    }
-
-    fun getShapeMObject(): String {
-        return "$ident.shape"
-    }
+interface StyleableShape {
+    fun restyle(styleProperties: StylesheetProperty): List<String>
 }
 
 class Rectangle(
     override val ident: String,
     override val text: String,
     color: String? = null,
     textColor: String? = null,
-    textWeight: String? = null,
-    font: String? = null,
-) : ShapeWithText() {
+) : ShapeWithText(), StyleableShape {
     override val classPath: String = "python/rectangle.py"
     override val className: String = "Rectangle_block"
     override val pythonVariablePrefix: String = "rectangle"
 
     init {
         color?.let { style.addStyleAttribute(Color(it)) }
         textColor?.let { style.addStyleAttribute(TextColor(it)) }
-        textWeight?.let { style.addStyleAttribute(TextWeight(it)) }
-        font?.let { style.addStyleAttribute(Font(it)) }
     }
+
+    override fun restyle(styleProperties: StylesheetProperty): List<String> {
+        val instructions = mutableListOf<String>()
+
+        if (styleProperties.borderColor != null) {
```

Comment:
Can these null checks be done in a let statement. They are more concise

Reply:
Can you show me the documentation in kotlin that shows this is how to do null checks using let? All I see is a lambda with the same variable inside?

Reply:
It is done in the init function above this but `styleProperties.borderColor?.let {instructions.add(...) }` is an example

Reply:
https://kotlinlang.org/docs/reference/null-safety.html in the safe call section